core/dbt/adapters/factory.py

```diff
@@ -85,8 +85,9 @@ def load_plugin(self, name: str) -> Type[Credentials]:
 
         return plugin.credentials
 
-    def register_adapter(self, config: AdapterRequiredConfig) -> None:
-        adapter_name = config.credentials.type
+    def register_adapter(self, config: AdapterRequiredConfig, adapter_name: Optional[str] = None) -> None:
+        if adapter_name is None:
```

Comment:
Should we add a single line comment describing that this behavior is mainly for Python-engines?